python/task_manager/results_poller.py

```diff
@@ -9,103 +9,179 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 """
-Results queue poller
+Results dispatcher for the background task manager.
 """
 
 import Queue
 from sgtk.platform.qt import QtCore
 import sgtk
 
 
-class ResultsPoller(QtCore.QObject):
+class TaskCompletedEvent(object):
     """
-    Polls a queue which holds the results posted from the worker threads.
+    Event sent when a task is succesfully completed.
     """
 
+    def __init__(self, worker_thread, task, result, results_queue):
+        """
+        Constructor.
+
+        :param worker_thread: Worker thread that completed sucessfully.
+        :param task: Task id of the completed task.
+        :param result: Result published by the task.
+        :param results_queue: Results queue to signal task completion.
+        """
+        self._worker_thread = worker_thread
+        self._task = task
+        self._result = result
+        self._results_queue = results_queue
+
+    def execute(self):
+        """
+        Signals the task_completed event on the results queue.
+        """
+        self._results_queue.task_completed.emit(self._worker_thread, self._task, self._result)
+
+
+class TaskFailedEvent(object):
+    """
+    Event sent when a task is succesfully completed.
+    """
+
+    def __init__(self, worker_thread, task, msg, traceback_, results_queue):
+        """
+        Constructor.
+
+        :param worker_thread: Worker thread that completed sucessfully.
+        :param task: Task id of the completed task.
+        :param msg: Error message from the worker thread.
+        :param traceback_: Traceback from  the worker thread error.
+        :param results_queue: Results queue to signal task failure.
+        """
+        self._worker_thread = worker_thread
+        self._task = task
+        self._msg = msg
+        self._traceback = traceback_
+        self._results_queue = results_queue
+
+    def execute(self):
+        """
+        Signals the task_failed event on the results queue.
+        """
+        self._results_queue.task_failed.emit(self._worker_thread, self._task, self._msg, self._traceback)
+
+
+class ResultsDispatcher(QtCore.QThread):
+    """
+    Dispatches events synchronously to the thread that owns this object.
+
+    Signalling between two different threads in PySide is broken in several versions
```

Comment:
This blurb used to be in the public documentation of the API because the polling had an impact on performance. Now that events are processed as soon a the main thread receives them it means that the dispatcher thread is just an implementation detail and it can be omitted from the documentation. As far as users are concerned, nothing has changed functionally from 4.1.0.